File: DrawingModel/DrawingInCSharp/EquilateralTriangle.cs
using System;

namespace DrawingCSharp
{
    public class EquilateralTriangle : IShape, IRotatable
    {
        double side = 0;
        double orientation = 0;

        public EquilateralTriangle(double side)
        {
            this.side = side;
        }

        public void GrowBy(double percent)
        {
            side = side * (1 + percent / 100);
        }

        public string Summary()
        {
            return "Equilateral Triangle, side: " + side + " orientation:" + orientation;
        }

        public void RotateBy(int degrees)
        {
            orientation = (orientation + degrees) % 360;
        }

        public byte[] DrawAsBitMap()
        {
            throw new NotImplementedException();
        }
    }
}

File: DrawingModel/DrawingInVB/EquilateralTriangle.vb


Namespace DrawingCSharp
	Public Class EquilateralTriangle
		Implements IShape
		Implements IRotatable
		Private side As Double = 0
		Private orientation As Double = 0

		Public Sub New(side As Double)
			Me.side = side
		End Sub

        Public Sub GrowBy(percent As Double) Implements IShape.GrowBy
            side = side * (1 + percent / 100)
        End Sub

        Public Function Summary() As String Implements IShape.Summary
            Return "Equilateral Triangle, side: " + side.ToString() + " orientation:" + orientation.ToString()
        End Function

        Public Sub RotateBy(degrees As Integer) Implements IRotatable.RotateBy
            orientation = (orientation + degrees) Mod 360
        End Sub

        Public Function DrawAsBitMap() As Byte() Implements IShape.DrawAsBitMap
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================
